interceptors/payment_authentication_test.go

```diff
@@ -3,72 +3,214 @@ package interceptors
+	"github.com/companieshouse/payments.api.ch.gov.uk/dao"
 	"github.com/companieshouse/payments.api.ch.gov.uk/helpers"
 	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+	"github.com/companieshouse/payments.api.ch.gov.uk/service"
+	"github.com/golang/mock/gomock"
 	"github.com/gorilla/mux"
+	"gopkg.in/jarcoal/httpmock.v1"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
 	. "github.com/smartystreets/goconvey/convey"
 )
 
+var defaultCost = models.CostResource{
+	Amount:                  "10",
+	AvailablePaymentMethods: []string{"method"},
+	ClassOfPayment:          []string{"class"},
+	Description:             "desc",
+	DescriptionIdentifier:   "identifier",
+	Links: models.Links{Self: "self"},
+}
+
+var defaultCostArray = []models.CostResource{
+	defaultCost,
+}
+
+func createMockPaymentService(dao *dao.MockDAO, config *config.Config) service.PaymentService {
+	return service.PaymentService{
+		DAO:    dao,
+		Config: *config,
+	}
+}
+
+func createMockPaymentPaymentAuthenticationInterceptor(service *service.PaymentService) PaymentAuthenticationInterceptor {
+	return PaymentAuthenticationInterceptor{
+		Service: *service,
+	}
+}
+
 func TestUnitUserPaymentInterceptor(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	cfg, _ := config.Get()
+	cfg.DomainWhitelist = "http://dummy-resource"
 
 	Convey("No payment ID in request", t, func() {
 		path := fmt.Sprintf("/payments/")
-		req, err := http.NewRequest("Get", path, nil)
+		req, err := http.NewRequest("GET", path, nil)
 		req.Header.Set("Eric-Identity", "authorised_identity")
 		req.Header.Set("Eric-Identity-Type", "oauth2")
 		req.Header.Set("ERIC-Authorised-User", "test@test.com;test;user")
-		req.Header.Set("ERIC-Authorised-Roles", "notauth2")
+		req.Header.Set("ERIC-Authorised-Roles", "noroles")
 		So(err, ShouldBeNil)
 
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mockPaymentAuthenticationInterceptor := createMockPaymentPaymentAuthenticationInterceptor(&mockPaymentService)
+
 		w := httptest.NewRecorder()
-		test := PaymentAuthenticationInterceptor(GetTestHandler())
+		test := mockPaymentAuthenticationInterceptor.PaymentAuthenticationIntercept(GetTestHandler())
 		test.ServeHTTP(w, req)
 		So(w.Code, ShouldEqual, 400)
 	})
 
 	Convey("Invalid user details in context", t, func() {
 		path := fmt.Sprintf("/payments/%s", "1234")
-		req, err := http.NewRequest("Get", path, nil)
+		req, err := http.NewRequest("GET", path, nil)
 		req = mux.SetURLVars(req, map[string]string{"payment_id": "1234"})
 		req.Header.Set("Eric-Identity", "authorised_identity")
 		req.Header.Set("Eric-Identity-Type", "oauth2")
 		req.Header.Set("ERIC-Authorised-User", "test@test.com;test;user")
-		req.Header.Set("ERIC-Authorised-Roles", "notauth2")
+		req.Header.Set("ERIC-Authorised-Roles", "noroles")
 		// The details have to be in a authUserDetails struct, so pass a different struct to fail
```

Comment:
I'm assuming the Eric headers just have to have something in them at this point as opposed to meaningful data?